### PR TITLE
klientctl: add bash autocomplete for machines and mounts.

### PR DESCRIPTION
**Pull request by ppknap**
~Depends on #11314~

### CODE REVIEW

File: go/src/koding/klient/machine/machinegroup/id.go [hunk 1]
```diff
@@ -4,6 +4,8 @@ import (
 	"errors"
 
 	"koding/klient/machine"
+	"koding/klient/machine/machinegroup/idset"
+	"sort"
```
Comment (rjeczalik) [inline go/src/koding/klient/machine/machinegroup/id.go]: import grouping

File: go/src/koding/klient/machine/machinegroup/id.go [hunk 2]
```diff
@@ -45,3 +47,66 @@ func (g *Group) ID(req *IDRequest) (*IDResponse, error) {
...
+
+	if req.IPs {
+		for _, id := range regAddress {
+			if ipAddr, err := g.address.Latest(id, "ip"); err == nil {
+				identifiers = append(identifiers, ipAddr.Value)
+			}
+		}
+	}
+
+	sort.Strings(identifiers)
```
Comment (rjeczalik) [inline go/src/koding/klient/machine/machinegroup/id.go]: 👍

File: go/src/koding/klient/machine/machinegroup/mount.go [hunk 3]
```diff
@@ -14,6 +14,7 @@ import (
 	"koding/klient/machine/mount"
 	"koding/klient/machine/mount/prefetch"
 	"koding/klient/machine/mount/sync/history"
+	"sort"
```
Comment (rjeczalik) [inline go/src/koding/klient/machine/machinegroup/mount.go]: ditto

File: go/src/koding/klientctl/commands/autocomplete.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        done
+
+        COMPREPLY=( "${COMPREPLY[@]}" $( compgen -W "${words[*]}" -- "$cur" ) )
+    fi
+}
+
+# This command provides auto completion abilities for exec subcommand.
+__kd_exec_completion()
+{
+    if [[ ${prev} == "exec" ]]; then
```
Comment (rjeczalik) [inline go/src/koding/klientctl/commands/autocomplete.go]: Out of curiosity - where's prev defined?

**Review by rjeczalik — APPROVED** [review]

File: go/src/koding/klientctl/commands/autocomplete.go (hunk 4 above)
Comment (ppknap) [inline go/src/koding/klientctl/commands/autocomplete.go]: This var is a standard name that holds last token. It's generated by cobra in its main function:

```bash
__start_kd()
{
    local cur prev words cword
    declare -A flaghash 2>/dev/null || :
    if declare -F _init_completion >/dev/null 2>&1; then
        _init_completion -s || return
    else
        __my_init_completion -n "=" || return
    fi

    local c=0
    local flags=()
    local two_word_flags=()
    local local_nonpersistent_flags=()
    local flags_with_completion=()
    local flags_completion=()
    local commands=("kd")
    local must_have_one_flag=()
    local must_have_one_noun=()
    local last_command
    local nouns=()

    __handle_word
}

if [[ $(type -t compopt) = "builtin" ]]; then
    complete -o default -F __start_kd kd
else
    complete -o default -o nospace -F __start_kd kd
fi

```

**Review by cihangir — APPROVED** [review]

**Pull request merged**